cmd/tendermint/commands/init.go

```diff
@@ -91,21 +105,43 @@ func initFilesWithConfig(config *cfg.Config) error {
 		ctx, cancel := context.WithTimeout(context.TODO(), ctxTimeout)
 		defer cancel()
 
-		pubKey, err := pv.GetPubKey(ctx)
-		if err != nil {
-			return fmt.Errorf("can't get pubkey: %w", err)
+		// if this is a validator we add it to genesis
+		if pv != nil {
+			pubKey, err := pv.GetPubKey(ctx)
+			if err != nil {
+				return fmt.Errorf("can't get pubkey: %w", err)
+			}
+			genDoc.Validators = []types.GenesisValidator{{
+				Address: pubKey.Address(),
+				PubKey:  pubKey,
+				Power:   10,
+			}}
 		}
-		genDoc.Validators = []types.GenesisValidator{{
-			Address: pubKey.Address(),
-			PubKey:  pubKey,
-			Power:   10,
-		}}
 
 		if err := genDoc.SaveAs(genFile); err != nil {
 			return err
 		}
 		logger.Info("Generated genesis file", "path", genFile)
 	}
 
+	// write config file
+	cfg.WriteConfigFile(config.RootDir, config)
+	logger.Info("Generated config", "mode", config.Mode)
+
 	return nil
 }
+
+func modePrompt() string {
```

Comment:
I think it'd be better to just error with a helpful message rather than do something interactive (particularly if we don't do interactive things in other contexts.) 

A good error message is easy to handle and is sort of expected in some situations (in ways that hanging forever may not be easy to understand) If we're really committed to having the prompt, then maybe having it timeout or error after a certain number of attempts?

Reply:
Yeah I don't mind replacing the prompt with an error. I just want to make it easy for people to transition from the previous behavior to this new one